server/src/models/User.js

```diff
@@ -64,10 +65,20 @@ userSchema.methods.getProfile = function() {
   }
 };
 
+userSchema.methods.getAgencyId = function(requestBody) {
+  if(this.role === 'developer') return mongoose.Types.ObjectId(requestBody.agency);
+
+  return this.agency;
+};
+
+userSchema.methods.getAgency = function() {
+  return Agency.findById(this.agency);
+};
+
 userSchema.statics.findByToken = function(token) {
   try {
     const user = untokenize(token);
-    return Promise.resolve(user);
+    return Promise.resolve(User.hydrate(user));
```

Comment:
`hydrate` the user here so things are easier in other files.

https://mongoosejs.com/docs/api.html#model_Model.hydrate